File: app/channels/chat_channel.rb
class ChatChannel < ApplicationCable::Channel
  def subscribed
    stream_from "secretroom_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def send_message(data)
    ActionCable.server.broadcast "secretroom_channel", data
  end
end

File: app/javascript/channels/chat_channel.js
import consumer from "channels/consumer"

window.chat_app = consumer.subscriptions.create("ChatChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
    window.appendThread(data);
  },

  send_message: function(data) {
    return this.perform('send_message', data);
  }
});
